File: server/src/protocol/GuardStatusCommand.cpp
#include "GuardStatusCommand.h"
#include "CommandVisitor.h"

using namespace std;
using namespace communication;

GuardStatusCommand::GuardStatusCommand()
        : Command(10, CommandType::GUARD_STATUS)
{}

GuardStatusCommand::~GuardStatusCommand()
{}

vector<uint8_t> GuardStatusCommand::getFrameBytes()
{
    initializeDataSize();

    vector<uint8_t > frame = getHeader();
    frame.push_back(static_cast<uint8_t>(commandType_));

    return frame;
}

std::string GuardStatusCommand::getName()
{
    return string("GuardStatusCommand");
}

void GuardStatusCommand::accept(CommandVisitor &visitor)
{
    visitor.visit(*this);
}

void GuardStatusCommand::initializeDataSize()
{
    uint16_t dataSize = sizeof(commandType_);

    setDataSize(dataSize);
}

File: mobile_client/app/src/main/java/wpam/mobile_client/protocol/GuardStatusCommand.java
package wpam.mobile_client.protocol;
import java.util.ArrayList;

public final class GuardStatusCommand extends Command
{
	public GuardStatusCommand()
	{
		super(10, CommandType.GUARD_STATUS_COMMAND);
	}
	public final void close()
	{
		super.close();
	}

	@Override
	public ArrayList<Integer> getFrameBytes()
	{
		initializeDataSize();

		ArrayList<Integer> frame = getHeader();
		frame.add(commandType_.getValue());

		return frame;
	}
	@Override
	public String getName()
	{
		return new String("GuardStatusCommand");
	}
	@Override
	public void accept(CommandVisitor visitor)
	{
		visitor.visit(this);
	}
	@Override
	protected void initializeDataSize()
	{
		setDataSize(1);
	}
}